src/modules/flow/http-server/http-server.c

```diff
@@ -393,7 +479,13 @@ static void
 common_close(struct sol_flow_node *node, void *data)
 {
     struct http_data *mdata = data;
+    uint16_t i;
+    struct sol_http_progressive_response *sse_client;
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&mdata->sse_clients, sse_client, i)
```

Comment:
not expected in the current implementation, but it is better to steal the vector and zero the existing before you dispatch these, so if the "on_close" callback fires it won't affect the vector you're iterating. Then you iterate and clear the copy you stored locally in this function